### ISSUE TITLE
patch not reversible

### ISSUE THREAD

**Issue by Thomas-git** [sonnyp/JSON8#65]
take this input :
```JSON
[{"nom":"joe","age":26},{"age":30,"nom":"arnaud"},{"age":26,"nom":"émilie"}]
```
apply this patch :
```JSON
[{"op":"remove","path":"/1"},{"op":"move","from":"/0/age","path":"/1/age"},{"op":"add","path":"/-","value":"essai"}]
```
the revert patch produced by `buildRevertPatch` do not revert data.
In fact, move operation overwrite an existing value and `revert` fail to restore it.

**Comment by sonnyp** [comment]
Test case:

```js
const ooPatch = require('./packages/patch')
const ooClone = require('./packages/json8/lib/clone')

const doc = [{ "nom": "joe", "age": 26 }, { "age": 30, "nom": "arnaud" }, { "age": 26, "nom": "émilie" }]
const patch = [{ "op": "remove", "path": "/1" }, { "op": "move", "from": "/0/age", "path": "/1/age" }, { "op": "add", "path": "/-", "value": "essai" }]

const applyResult = ooPatch.apply(ooClone(doc), patch, { reversible: true })

const revertPatch = ooPatch.buildRevertPatch(applyResult.revert)
console.log(doc)
console.log(ooPatch.apply(applyResult.doc, revertPatch).doc)
```

**Comment by Thomas-git** [comment]
That's it. 'age' is missing for nom:"émilie"

**Comment by sonnyp** [comment]
on it :)

**Comment by sonnyp** [comment]
fixed in json8-patch 0.4.2

**Comment by sonnyp** [comment]
thanks for reporting the issue, see https://github.com/sonnyp/JSON8/pull/77
